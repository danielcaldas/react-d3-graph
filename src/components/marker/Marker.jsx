import React from "react";

/**
 * Market component provides configurable interface to marker definition.
 * @example
 *
 * <Marker id="marker-id" fill="black" />
 */
export default class Marker extends React.Component {
    render() {
        return (
            <marker
                className="marker"
                id={this.props.id}
                viewBox="0 -5 10 10"
                refX={this.props.refX}
                refY="0"
                markerWidth={this.props.markerWidth}
                markerHeight={this.props.markerHeight}
                orient="auto"
                fill={this.props.fill}
            >
                <path d="M0,-5L10,0L0,5" />
            </marker>
        );
    }
}
