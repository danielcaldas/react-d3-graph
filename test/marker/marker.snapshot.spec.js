import React from "react";
import renderer from "react-test-renderer";

import Marker from "../../src/components/marker/Marker";

describe("Snapshot - Marker Component", () => {
    let that = {};

    beforeEach(() => {
        that.clickCallback = jest.fn();
        that.mouseOverCallback = jest.fn();
        that.mouseOutCallback = jest.fn();

        that.marker = renderer.create(<Marker id="id" refX="5" fill="green" />);

        that.tree = that.marker.toJSON();
    });

    test("should match snapshot", () => {
        expect(that.tree).toMatchSnapshot();
    });
});
