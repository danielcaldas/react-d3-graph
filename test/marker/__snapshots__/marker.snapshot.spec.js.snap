// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot - Marker Component should match snapshot 1`] = `
<marker
  className="marker"
  fill="green"
  id="id"
  orient="auto-start-reverse"
  refX="5"
  refY="0"
  viewBox="0 -5 10 10"
>
  <path
    d="M0,-5L10,0L0,5"
  />
</marker>
`;
